mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -388,19 +388,21 @@ class Word2Vec extends Serializable with Logging {
         val syn0Local = model._1
         val syn1Local = model._2
         // Only output modified vectors.
-        Iterator.tabulate(vocabSize) { index =>
+        val modifiedSize = syn0Modify.count(_ > 0) + syn1Modify.count(_ > 0)
+        val modified = new Array[(Int, Array[Float])](modifiedSize)
+        var mi = 0
+        (0 until vocabSize).foreach { index =>
```

Comment:
Ping @hhbyyh I suppose this is OK since the resulting RDD then calls `reduceByKey`, to add the vectors, so order should not matter.